### PR TITLE
doc: modules: document recommendations

### LINKED ISSUE #49766: Document downstream module configuration recommendations
**Is your enhancement proposal related to a problem? Please describe.**
Downstream module developers need best practice recommendations on how to organize 
and configure their modules to best align with the Zephyr build and configuration system.
These types of recommendations are lacking in the current Zephyr documentation.

**Describe the solution you'd like**
Extend the Zephyr module documentation with the following recommendations:

1. Update the Module Name section of the Modules documentation to recommend that the name be set.  
Further indicate that this is necessary to prevent inadvertent name changes through a series of west manifest imports.
2. Add a new section to the Modules documentation recommending all the public headers defined by a module should be exposed in the include file search list with a pathname beginning with a module-name directory (e.g., mcuboot public bootutil/bootutil.h header is included as "mcuboot/bootutil/bootutil.h").

**Describe alternatives you've considered**
I alternatively considered adding a new section specifically for downstream modules, but these recommendations should be applied to the Zephyr repository as well.

**Additional context**

### PR DESCRIPTION
**Pull request by gregshue**
Update Modules (External projects) documentation to:

- Recommend explicitly setting the module name in module.yml
- Recommend exposing module-provided headers with an include
  pathname beginning with the module-name.

Fixes #49766

Signed-off-by: Gregory SHUE <gregory.shue@legrand.com>

### CODE REVIEW

**Comment by tejlmand** [comment]
please remove merge commit.

I believe this is nice improvement, but wonder if it should be discussed at process WG @carlescufi / @mbolivar-nordic  ?

**Comment by tejlmand** [comment]
> @tejlmand The issue and PR originated from an action item from a discussion in the process working group on 8/31.

Great, then just get rid of the merge commit 2aefeec92f75d96f0e4c0dac8d14c4d8c923d7c3 and i'll give it a +1.

**Comment by stephanosio** [comment]
Rebased

File: doc/develop/modules.rst [hunk 1]
```diff
@@ -58,6 +58,8 @@ Module Repositories
      Existing module repositories that do not conform to the above convention
      do not need to be renamed to comply with the above convention.
 
+* Module repositories names should be explicitly set in the *module.yml* file.
```
Comment (stephanosio) [inline doc/develop/modules.rst:61]:
```suggestion
* Module repositories names should be explicitly set in the :file:`zephyr/module.yml` file.
```

File: doc/develop/modules.rst [hunk 2]
```diff
@@ -428,7 +434,9 @@ Module name
 Each Zephyr module is given a name by which it can be referred to in the build
 system.
 
-The name may be specified in the :file:`zephyr/module.yml` file:
+The name should be specified in the :file:`zephyr/module.yml` file. This will
+ensure the module name is not changable through user-defined directory names
```
Comment (stephanosio) [inline doc/develop/modules.rst:438]: typo

```suggestion
ensure the module name is not changeable through user-defined directory names
```

**Review by stephanosio — COMMENTED** [review]
need to squash the revision commits into the first commit

**Review by tejlmand — APPROVED** [review]
lgtm

**Review by stephanosio — APPROVED** [review]

**Pull request merged**